packages/react/slider/src/Slider.tsx

```diff
@@ -486,28 +488,44 @@ SliderRange.displayName = RANGE_NAME;
 const THUMB_NAME = 'SliderThumb';
 const THUMB_DEFAULT_TAG = 'span';
 
-type SliderThumbOwnProps = Omit<Polymorphic.OwnProps<typeof SliderThumbImpl>, 'value' | 'index'>;
+type SliderThumbOwnProps = Omit<Polymorphic.OwnProps<typeof SliderThumbImpl>, 'index'>;
 type SliderThumbPrimitive = Polymorphic.ForwardRefComponent<
   Polymorphic.IntrinsicElement<typeof SliderThumbImpl>,
   SliderThumbOwnProps
 >;
 
 const SliderThumb = React.forwardRef((props, forwardedRef) => {
-  const { ref: collectionRef, index } = useSliderCollectionItem();
-  const ref = useComposedRefs(forwardedRef, collectionRef);
-  const context = useSliderContext(THUMB_NAME);
-  const value = context.values[index];
-  return value !== undefined ? (
-    <SliderThumbImpl {...props} ref={ref} index={index} value={value} />
+  const { getItems } = useCollection();
+  const [thumb, setThumb] = React.useState<React.ElementRef<typeof SliderThumbImpl> | null>(null);
+  const composedRefs = useComposedRefs(forwardedRef, (node) => setThumb(node));
+  const isInitialRenderRef = React.useRef(true);
+  const index = React.useMemo(
+    () => (thumb ? getItems().findIndex((item) => item.ref.current === thumb) : -1),
+    [getItems, thumb]
+  );
+
+  React.useEffect(() => {
+    isInitialRenderRef.current = false;
+  }, []);
+
+  /**
+   * We hide thumbs on the first render while we work out the index, otherwise SSR will
+   * render them in the wrong position before they snap into the correct position during
+   * hydration which would be visually jarring.
```

Comment:
Might have been ok to leave it as is rather than doing more work specifically to hide it?
Are you thinking that we'll deal with this anyway differently at some point?

Reply:
You think it's okay for them to render in the wrong position and snap into a different position during hydration?

The plan is to fix the SSR issue eventually so that it renders in the correct position server-side.

Reply:
It’s not great but I wonder how visible it is in a real use case. But yeah regardless SSR is something we need to look at some point. We might end up with some controlled overrides (like providing the index yourself) when you want to make sure it’s ok on first server render.

Reply:
I think these kind of solutions would definitely be the easiest. Simple trade offs.